test/cases/api.sh

```diff
@@ -198,73 +335,106 @@ do
   if [[ "$COMPOSE_STATUS" != "pending" && "$COMPOSE_STATUS" != "running" ]]; then
     test "$COMPOSE_STATUS" = "success"
     test "$UPLOAD_STATUS" = "success"
-    test "$UPLOAD_TYPE" = "aws"
+    # Do not check the return type for now, since it is not returned from cloudapi
+    # test "$UPLOAD_TYPE" = "$CLOUD_PROVIDER"
+    test "$UPLOAD_TYPE" = ""
     break
   fi
 
   sleep 30
 done
 
-
 #
-# Verify the image landed in EC2, and delete it.
+# Verify the image landed in the appropriate cloud provider, and delete it.
 #
 
-$AWS_CMD ec2 describe-images \
+# Verify image in EC2 on AWS
+function verifyInAWS() {
+  $AWS_CMD ec2 describe-images \
     --owners self \
-    --filters Name=name,Values="$SNAPSHOT_NAME" \
+    --filters Name=name,Values="$AWS_SNAPSHOT_NAME" \
     > "$WORKDIR/ami.json"
 
-AMI_IMAGE_ID=$(jq -r '.Images[].ImageId' "$WORKDIR/ami.json")
-SNAPSHOT_ID=$(jq -r '.Images[].BlockDeviceMappings[].Ebs.SnapshotId' "$WORKDIR/ami.json")
-SHARE_OK=1
+  AMI_IMAGE_ID=$(jq -r '.Images[].ImageId' "$WORKDIR/ami.json")
+  AWS_SNAPSHOT_ID=$(jq -r '.Images[].BlockDeviceMappings[].Ebs.SnapshotId' "$WORKDIR/ami.json")
+  SHARE_OK=1
 
-# Verify that the ec2 snapshot was shared
-$AWS_CMD ec2 describe-snapshot-attribute --snapshot-id "$SNAPSHOT_ID" --attribute createVolumePermission > "$WORKDIR/snapshot-attributes.json"
+  # Verify that the ec2 snapshot was shared
+  $AWS_CMD ec2 describe-snapshot-attribute --snapshot-id "$AWS_SNAPSHOT_ID" --attribute createVolumePermission > "$WORKDIR/snapshot-attributes.json"
 
-SHARED_ID=$(jq -r '.CreateVolumePermissions[0].UserId' "$WORKDIR/snapshot-attributes.json")
-if [ "$AWS_API_TEST_SHARE_ACCOUNT" != "$SHARED_ID" ]; then
+  SHARED_ID=$(jq -r '.CreateVolumePermissions[0].UserId' "$WORKDIR/snapshot-attributes.json")
+  if [ "$AWS_API_TEST_SHARE_ACCOUNT" != "$SHARED_ID" ]; then
     SHARE_OK=0
-fi
+  fi
 
-# Verify that the ec2 ami was shared
-$AWS_CMD ec2 describe-image-attribute --image-id "$AMI_IMAGE_ID" --attribute launchPermission > "$WORKDIR/ami-attributes.json"
+  # Verify that the ec2 ami was shared
+  $AWS_CMD ec2 describe-image-attribute --image-id "$AMI_IMAGE_ID" --attribute launchPermission > "$WORKDIR/ami-attributes.json"
 
-SHARED_ID=$(jq -r '.LaunchPermissions[0].UserId' "$WORKDIR/ami-attributes.json")
-if [ "$AWS_API_TEST_SHARE_ACCOUNT" != "$SHARED_ID" ]; then
+  SHARED_ID=$(jq -r '.LaunchPermissions[0].UserId' "$WORKDIR/ami-attributes.json")
+  if [ "$AWS_API_TEST_SHARE_ACCOUNT" != "$SHARED_ID" ]; then
     SHARE_OK=0
-fi
-
-# Create key-pair
-$AWS_CMD ec2 create-key-pair --key-name "key-for-$AMI_IMAGE_ID" --query 'KeyMaterial' --output text > keypair.pem
```

Comment:
I think you removed this step causing the CI to fail.

Reply:
Yeah, not sure how that happened. Thanks Ondrej!

Reply:
This is fixed now...